### PR TITLE
feat(searchBox): allow to pass a queryHook

### PR DESCRIPTION
**Pull request by vvo**
The queryHook can be used to implement debouncing for example.

Usage:

```js
searchBox({
  queryHook: (originalQuery, search) => // either call search() or search(newQuery);
});
```

### CODE REVIEW

File: src/widgets/search-box/search-box.js [hunk 1]
```diff
@@ -137,26 +147,33 @@ function searchBox({
...
       }
 
-      function setQuery(e) {
-        helper.setQuery(getValue(e));
+
+      function maybeSearch(originalQuery) {
+        if (queryHook) {
+          queryHook(
+            originalQuery,
+            newQuery => newQuery !== undefined ? search(newQuery) : search(originalQuery)
```
Comment (bobylito) [inline src/widgets/search-box/search-box.js]: Less magic if we were to let the user pass the original query himself.

Comment (vvo) [inline src/widgets/search-box/search-box.js]: Agreed, changing that

File: src/widgets/search-box/search-box.js [hunk 2]
```diff
@@ -168,6 +185,10 @@ function searchBox({
         return actualEvent => actualEvent.keyCode === expectedKeyCode && fnToExecute(actualEvent);
       }
 
+      function getQueryAndCall(fnToExecute) {
```
Comment (bobylito) [inline src/widgets/search-box/search-box.js]: fnToExecute could be reduced to `f` or `fn`. Also those functional helpers :100: could be moved somewhere else.

Comment (vvo) [inline src/widgets/search-box/search-box.js]: I renamed to "func" and moved them at the bottom of the file for now to indicate that there's no shared state

Comment (vvo) [inline src/widgets/search-box/search-box.js]: If we reuse them somewhere we can move them

**Comment by bobylito** [comment]
Nice! LGTM!

**Pull request merged**